internal/pkg/logs/task.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package logs
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/copilot-cli/internal/pkg/aws/cloudwatchlogs"
+	"github.com/aws/copilot-cli/internal/pkg/aws/ecs"
+	"github.com/aws/copilot-cli/internal/pkg/task"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+)
+
+const (
+	numCWLogsCallsPerRound = 10
+	fmtTaskLogGroupName    = "/copilot/%s"
+)
+
+// TasksDescriber describes ECS tasks.
+type TasksDescriber interface {
+	DescribeTasks(cluster string, taskARNs []string) ([]*ecs.Task, error)
+}
+
+// EventsLogger gets a log group's log events.
+type EventsLogger interface {
+	TaskLogEvents(logGroupName string,
+		streamLastEventTime map[string]int64,
+		opts ...cloudwatchlogs.GetLogEventsOpts) (*cloudwatchlogs.LogEventsOutput, error)
+}
+
+// TaskLogs represents a service that writes task log events.
+type TaskLogs struct {
+	GroupName string
+	Tasks     []*task.Task
+
+	Writer       io.Writer
+	EventsLogger EventsLogger
+	Describer    TasksDescriber
+
+	// Fields that are private that get modified each time
+	runningTasks []*task.Task
+}
+
+// NewTaskLogs returns a TaskLogs configured against the input.
+func NewTaskLogs(groupName string, tasks []*task.Task, sess *session.Session) *TaskLogs {
```

Comment:
nit: Can we have `session` as the first parameter or maybe the interface for a session: https://docs.aws.amazon.com/sdk-for-go/api/aws/client/#ConfigProvider to be consistent with the constructors of the AWS SDK.